parameters_meta_schema.json

```diff
@@ -1,155 +1,316 @@
+                        },
+                        "description": {
+                            "type": "string",
+                            "description": "NON STANDARD OPTION: The description of the currrent parameter"
+                        },
+                        "help_text": {
+                            "type": "string",
+                            "description": "NON STANDARD OPTION: A more detailed help text"
+                        },
+                        "fa_icon": {
+                            "type": "string",
+                            "pattern": "^fa",
+                            "description": "NON STANDARD OPTION: A font awesome icon to use in the nf-core parameter documentation"
+                        },
+                        "errorMessage": {
+                            "type": "string",
+                            "minLength": 1,
+                            "description": "NON STANDARD OPTION: The custom error message to display in case validation against this parameter fails. Can also be used as a deprecation message if 'deprecated' is true"
+                        },
+                        "default": {
+                            "$ref": "#/$defs/allTypes",
+                            "description": "Specifies a default value to use for this parameter"
+                        },
+                        "examples": {
+                            "type": "array",
+                            "items": {
+                                "$ref": "#/$defs/allTypes"
+                            },
+                            "description": "A list of examples for the current parameter"
+                        },
+                        "deprecated": {
+                            "type": "boolean",
+                            "description": "States that the parameter is deprecated. Please provide a nice deprecation message using 'errorMessage'"
+                        },
+                        "hidden": {
+                            "type": "boolean",
+                            "description": "NON STANDARD OPTION: Hide this parameter from the help message and documentation"
+                        },
+                        "minLength": {
+                            "$ref": "#/$defs/nonNegativeInteger",
+                            "description": "The minimum length a 'string' parameter value should be"
+                        },
+                        "maxLength": {
+                            "$ref": "#/$defs/nonNegativeInteger",
+                            "description": "The maximum length a 'string' parameter value should be"
+                        },
+                        "minimum": {
+                            "type": "number",
+                            "description": "The mimimum value an 'integer' or 'number' parameter value should be"
```

Comment:
@nvnieuwk why do you need all the define these standard schema fields again?

Reply:
Because we only wanted to allow a subset of the real keywords and I didn't find another way to do this. This PR is not done yet though (I should maybe set this to draft). It's a pretty complex task, but it can probably be simplified a lot

Reply:
okay, I restructured it accordingly to make it more readable (commit coming soon)